packages/babel-plugin-transform-object-rest-spread/src/index.ts

```diff
@@ -526,37 +536,51 @@ export default declare((api, opts: Options) => {
         if (leftPath.isObjectPattern() && hasRestElement(leftPath)) {
           const nodes = [];
 
-          const refName = path.scope.generateUidBasedOnNode(
-            path.node.right,
-            "ref",
-          );
+          const { right } = path.node;
 
-          nodes.push(
-            t.variableDeclaration("var", [
-              t.variableDeclarator(t.identifier(refName), path.node.right),
-            ]),
-          );
+          let refName;
+          if (t.isIdentifier(right) && path.scope.hasBinding(right.name)) {
```

Comment:
Should we also check if the binding is constant? A getter in the destructured object could reassign it.

```js
let a = { get x() { a = { z: 2 } }, y: 2 };

let x, rest;
({ x, ...rest } = a);
```

Reply:
(add this as an exec test)